### DISCUSSION TITLE
Token expiration by client

### DISCUSSION THREAD

**Discussion by alexthelion** [keycloak/keycloak#8960] [Q&A]
I have multiple web clients with one users database (single realm). 
I want to set every client with different token expiration policy which afterwards user need to sign in again : 
i.e.: client a - 4 hours, client b - 90 days.
I'm using standard flow, from what I understand the only param that control the token expiration is **realm token setting**: SSO Session Max.
Tried to use client session max and it's not affecting token expiration.
Is there a way to achieve this?

**Comment by stianst** [comment]
Depends on what token you are talking about. Refresh token expiration is determined by SSO session, and client session, timeouts, while access token timeout has a global default, with an option to override for individual clients.

**Reply by alexthelion** [reply]
I'm trying to set a fixed token lifetime so the user should sign in again after its expiration. When I'm using the refresh token there is a long lifetime. How can i set a fixed lifetime by client?

**Reply by jbman** [reply]
Your client could have a client specific access token valid for 4 hours and request re-login when access token expired. Re-login can be enforced by the client with URL parameter `prompt=login` to the authorization endpoint.
